semgrep/semgrep/rule_match.py

```diff
@@ -246,6 +256,23 @@ def get_syntactic_id(self) -> str:
         hash_bytes = int.to_bytes(hash_int, byteorder="big", length=16, signed=False)
         return str(binascii.hexlify(hash_bytes), "ascii")
 
+    @pattern_based_id.default
+    def get_pattern_based_id(self) -> str:
```

Comment:
Need a comment explaining why this is better than simply using the rule ID.

Still think this is a bad name since it's actually analyzing multiple patterns in a formula, so formula_based_id would be better, but then it does not accomodate for the fact that we also using the metavars content, so it's a match_based_id.

Reply:
So do we want to switch the name to match_based_id? I'm ok with whatever

Reply:
Going to go ahead and rename it